### PR TITLE
CASMINST-4942 put Stage_1 in argo

### PR DESCRIPTION
**Pull request by leliasen-hpe**
# Description

Put stage_1 into argo.

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I don't have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by dzou-hpe — CHANGES_REQUESTED** [review]
procedure doc is missing

**Review by dzou-hpe — APPROVED** [review]

**Comment by rustydb** [comment]
@leliasen-hpe when I rebased and squashed this PR offline, it went well except for `storage.rebuild.yaml` which had lots of conflicts. That file is already in `main`, and I couldn't tell if yours or main's was "more right".

File: upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh [hunk 1]
```diff
@@ -35,25 +35,27 @@ retry=true
 force=false
 
 function usage() {
-    echo "CSM ncn worker upgrade script"
+    echo "CSM ncn worker and storage upgrade script"
     echo
     echo "Syntax: /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-nodes.sh [COMMA_SEPARATED_NCN_HOSTNAMES] [-f|--force|--retry|--base-url|--dry-run]"
     echo "options:"
-    echo "--no-retry     Do not automatically retry  (default: false)"
+    echo "--retry        Send RETRY request instead of CREATE if there is a failed worker rebuild/upgrade workflow already  (default: ${retry})"
```
Comment (dzou-hpe) [inline upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh:42]: @leliasen-hpe merge issue? we need to make retry default

File: upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh [hunk 2]
```diff
@@ -236,21 +258,21 @@ while true; do
         fi
 
         if [[ "${phase}" == "Failed" ]]; then
-            echo "Workflow in Failed state, Retry ..."
-            retryRebuildWorkflow "$workflow"
+            exit 1
+            break;
```
Comment (dzou-hpe) [inline upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh:262]: another merge issue

File: upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh [hunk 3]
```diff
@@ -236,21 +258,21 @@ while true; do
...
         fi
 
         if [[ "${phase}" == "Error" ]]; then
             echo "Workflow in Error state, Retry ..."
-            retryRebuildWorkflow "$workflow"
+            curl -sk -XPUT -H "Authorization: Bearer $(getToken)" "${baseUrl}/apis/nls/v1/workflows/${workflow}/retry"
```
Comment (dzou-hpe) [inline upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh:267]: here too

File: upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh [hunk 4]
```diff
@@ -236,21 +258,21 @@ while true; do
...
         fi
-        runningSteps=$(jq -jr ".[] | select(.name==\"${workflow}\") | .status.nodes[] | select(.type==\"Retry\")| select(.phase==\"Running\")  | .name + \"\n  \" " < "${res_file}")
-        succeededSteps=$(jq -jr ".[] | select(.name==\"${workflow}\") | .status.nodes[] | select(.type==\"Retry\")| select(.phase==\"Succeeded\")  | .name +\"\n  \" " < "${res_file}")
-        clear
+        runningSteps=$(jq -jr ".[] | select(.name==\"${workflow}\") | .status.nodes[] | select(.type==\"Retry\")| select(.phase==\"Running\")  | .displayName + \"\n  \" " < "${res_file}")
+        succeededSteps=$(jq -jr ".[] | select(.name==\"${workflow}\") | .status.nodes[] | select(.type==\"Retry\")| select(.phase==\"Succeeded\")  | .displayName +\"\n  \" " < "${res_file}")
         printf "\n%s\n" "Succeeded:"
-        echo "  ${succeededSteps}" | awk -F'.' '{print $2" -  "$3}'
+        echo "  ${succeededSteps}"
```
Comment (dzou-hpe) [inline upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh:272]: same here

**Review by dzou-hpe — CHANGES_REQUESTED** [review]

File: upgrade/Stage_1.md [hunk 5]
```diff
@@ -5,107 +5,18 @@
...
-    update_bss_storage
-    ```
+1. (`ncn-m001#`) Run `ncn-upgrade-worker-storage-nodes.sh` for `ncn-s001`.
+
+   ```bash
+   /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ncn-s001
+   ```
+
+   > **`NOTE`**
+   >> You can also upgrade multiple storage nodes with a comma separated list. This will upgrade the storage nodes sequentially.
```
Comment (GunashekarKM) [inline upgrade/Stage_1.md:15]:
```suggestion
   >> You can also upgrade multiple storage nodes with a comma-separated list. This upgrades the storage nodes sequentially.
```

**Review by GunashekarKM — COMMENTED** [review]
Review completed for the updated MD file content.

**Pull request closed without merging**